src/DetailsPage/Documents/Document.js

```diff
@@ -1,26 +1,26 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import { TableRow, TableColumn, FontIcon } from 'react-md';
 import { validateIntegrity } from './DocumentIntegrityValidator';
 
-class Document extends Component {
-  state = {
-    document: null,
-  };
+const Document = (props) => {
 
-  componentDidMount() {
-    this.validate(this.props.document);
-  }
+  const [document, setDocument] = useState(null);
 
-  componentWillReceiveProps(nextProps) {
-    this.validate(nextProps.document);
-  }
+  useEffect(() => {
+    validate(props.document);
+  }, [props.document])
 
-  validate = async document => {
+  const validate = async document => {
     const result = await validateIntegrity(document);
-    this.setState({ document: {...document, ...result} });
+    setDocument(prevDocument => {
+      return {
+        ...prevDocument,
```

Comment:
Initially you set the state of the `document` to null.
Here you use this null, and append two verification flags to if, while the document metadata get's lost. 

Please carefully check the original function
```
  validate = async document => {
    const result = await validateIntegrity(document);
    this.setState({ document: {...document, ...result} });
  }
```